Code/wwnet/network-typedefs.h

```diff
@@ -0,0 +1,13 @@
+#ifndef NETWORKTYPEDEFS_H
+#define NETWORKTYPEDEFS_H
+
+#ifdef _WINDOWS
+#include <winsock.h>
```

Comment:
I suggest we go all in using winsock2.h and link the ws2 library rather than winsock32 though both can supply the 2.2 version of the winsock API. You might want to add a few things from https://github.com/TheAssemblyArmada/Vanilla-Conquer/blob/vanilla/common/sockets.h as well.